radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/Atom.java

```diff
@@ -49,14 +50,19 @@
 	private final ImmutableMap<String, String> metaData;
 
 	private Atom() {
-		this.metaData = ImmutableMap.of();
+		this(0);
+	}
+
+	public Atom(long timestamp) {
+		this.metaData = ImmutableMap.of(METADATA_TIMESTAMP_KEY, String.valueOf(timestamp));
 	}
 
 	public Atom(List<ParticleGroup> particleGroups) {
```

Comment:
It's not possible to create an atom that will pass validation with this constructor.  I'm not sure it's necessary or useful.

Reply:
Fixed, good catch